sequencer/src/state_signature.rs

```diff
@@ -39,13 +48,36 @@ pub(super) async fn state_signature_loop<N>(
         } = event
         {
             if let Some(leaf) = leaf_chain.first() {
-                tracing::debug!("New leaves decided. Newest leaf: {:?}", leaf);
-                let new_state = form_light_client_state(leaf, stake_table_comm);
-                context.sign_new_state(&new_state);
+                let state = form_light_client_state(leaf, stake_table_comm);
+                let signature = context.sign_new_state(&state).await;
+                tracing::info!(
+                    "New leaves decided. Latest block height: {}, posting to relay server {:?}",
+                    leaf.get_height(),
+                    state_relay_server_url,
+                );
+
+                if let Some(state_relay_server_url) = state_relay_server_url {
+                    let client: Client<ServerError> = Client::new(state_relay_server_url.clone());
+                    client.connect(None).await;
```

Comment:
Would be good to create this client and `connect` once before entering the loop. You could create an `Option<Client>` at the top and then just `if let Some(client) = client_opt` to avoid reconnecting each time

Reply:
Does it mean that the server has to maintain many active connections? Although it might be okay for now.
Also will it reconnect if the connection is lost?

Reply:
addressed in [88a46f6](https://github.com/EspressoSystems/espresso-sequencer/pull/1016/commits/88a46f68497a8124094d2eba688e9a4cbf476815)